lib/matplotlib/hatch.py

```diff
@@ -181,6 +183,21 @@ def __init__(self, hatch, density):
     ]
 
 
+def _validate_hatch_pattern(hatch):
+    valid_hatch_patterns = '-+|/\\xXoO.*'
+    if hatch is not None:
+        invalid = next((c for c in hatch if c not in valid_hatch_patterns),
```

Comment:
Could be written using sets (if `valid_hatch_patterns` is also made a set):
```suggestion
        invalids = set(hatch).difference(valid_hatch_pattern)
```
Not sure if that's better. It could yield all invalid chars or you wrap it again with next(). This would at least fit in one line.

Reply:
This is probably better as it matches `validate_hatch` for rcParams (which we can use after the deprecation.)